src/l500/l535-amc-option.cpp

```diff
@@ -0,0 +1,85 @@
+//// License: Apache 2.0. See LICENSE file in root directory.
+//// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include "l535-amc-option.h"
+#include "l500-private.h"
+#include "l500-depth.h"
+
+
+using namespace librealsense;
+using namespace librealsense::ivcam2::l535;
+
+amc_option::amc_option( l500_device * l500_dev,
+                        hw_monitor * hw_monitor,
+                        amc_control type,
+                        const std::string & description )
+    : _device( l500_dev )
+    , _hw_monitor( hw_monitor )
+    , _control( type )
+    , _description( description )
+{
+    // Keep the USB power on while triggering multiple calls on it.
+    ivcam2::group_multiple_fw_calls( _device->get_depth_sensor(), [&]() {
+        auto min = _hw_monitor->send( command{ AMCGET, _control, get_min } );
+        auto max = _hw_monitor->send( command{ AMCGET, _control, get_max } );
+        auto step = _hw_monitor->send( command{ AMCGET, _control, get_step } );
+
+        if( min.size() < sizeof( int32_t ) || max.size() < sizeof( int32_t )
+            || step.size() < sizeof( int32_t ) )
+        {
+            std::stringstream s;
+            s << "Size of data returned is not valid min size = " << min.size()
+              << ", max size = " << max.size() << ", step size = " << step.size();
+            throw std::runtime_error( s.str() );
+        }
+
+        auto max_value = float( *( reinterpret_cast< int32_t * >( max.data() ) ) );
+        auto min_value = float( *( reinterpret_cast< int32_t * >( min.data() ) ) );
+
+        auto res = query_default();
+
+        _range = option_range{ min_value,
+                               max_value,
+                               float( *( reinterpret_cast< int32_t * >( step.data() ) ) ),
+                               res };
+    } );
+}
+
+float amc_option::query() const
+{
+    auto res = _hw_monitor->send( command{ AMCGET, _control, get_current, RS2_SENSOR_MODE_VGA } );
```

Comment:
Please comment why we're sending the sensor mode when it shouldn't be needed

Reply:
removed